### PR TITLE
cnf-tests: gatekeeper constraint validation test

### PR DESCRIPTION
**Pull request by mmirecki**
Add a basic gatekeeper constraint validation test.
If needed, additional constraints will be added in a follow up PR once this one is merged.

### CODE REVIEW

**Comment by SchSeba** [comment]
@mmirecki can you fix the commit message to make the CI happy?

**Comment by SchSeba** [comment]
@mmirecki I saw we are not running gatekeeper tests on the CI can we enable them first on a separate PR

**Comment by SchSeba** [comment]
```
<testcase name="gatekeeper mutation should be able to add metadata info(labels/annotations)" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper mutation should avoid mutating existing metadata info(labels/annotations)" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper mutation should apply mutations by order" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper mutation should be able to update mutation policy" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper mutation should not apply mutations policies after deletion" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper mutation should be able to match by any match category" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper constraints should apply constraints" classname="CNF Features e2e integration tests" time="0">
<skipped/>
</testcase>
<testcase name="gatekeeper operator should be able to select mutation namespaces" classname="CNF Features e2e integration tests" time="0">
<skipped/>
```

File: cnf-tests/testsuites/e2esuite/gatekeeper/gatekeeper.go [hunk 1]
```diff
@@ -695,6 +701,57 @@ var _ = Describe("gatekeeper", func() {
 		})
 	})
 
+	Context("constraints", func() {
+		AfterEach(func() {
+			err := deletePods(testingNamespace, client)
```
Comment (SchSeba) [inline cnf-tests/testsuites/e2esuite/gatekeeper/gatekeeper.go:706]: should this one point to the `constraintValidationNamespace` ?

File: cnf-tests/testsuites/e2esuite/test_suite_test.go [hunk 2]
```diff
@@ -31,14 +31,10 @@ import (
```
Comment (SchSeba) [inline cnf-tests/testsuites/e2esuite/test_suite_test.go:31]: I think you should not change this file

**Review by SchSeba — COMMENTED** [review]
small comments.

In general I will like the GK test to run before we merge this PR

File: cnf-tests/testsuites/e2esuite/gatekeeper/gatekeeper.go (hunk 1 above)
Comment (mmirecki) [reply cnf-tests/testsuites/e2esuite/gatekeeper/gatekeeper.go]: fixed

File: cnf-tests/testsuites/e2esuite/test_suite_test.go (hunk 2 above)
Comment (mmirecki) [reply cnf-tests/testsuites/e2esuite/test_suite_test.go]: fixed

**Pull request merged**